### DISCUSSION TITLE
Installer worked coding not giving result.

### DISCUSSION THREAD

**Discussion by pradeepsimba** [nokeedev/gradle-native#639] [General]
How to build qt shared library ?

[My project][111]

the app works perfectly if I build it as an app.

[TEMPLATE = app][112]

If I create a shared lib with same code the code not gives result.

I changed `TEMPLATE = app` to `TEMPLATE = lib` and I compile the same code. But , the result comes an error.

[code not give result as a qt shared lib][113]

[wifimouse][11] app fully build with qt.

[Source code of wifimouse server application][12]

I install and run wifimouse app in my computer it works perfectly and all the options are works smoothly.

But, If I compile the same code as a shared lib in my computer it not works properly.

For example ; In wifimouse application there is an option for brightness controling 

[Brightness code][13] 

[![enter image description here][14]][14]

error comes like this. But , it works smoothly as a [TEMPLATE = app][112].

Problem with qt in getting outputs.

[My project][1]

I want to get the cpu usage , ram usage ,process list and app list with qt.

**In Windows**

[![windows os][2]][2]

 I am getting ram usage from this [code][3].

 But [app list][4] , [cpu usage][5] and [process][6] outputs are not showed.

**In linux**

[![enter image description here][7]][7]

  
   In linux the [cpu usage][8] output not showed.

**In Mac**

[![enter image description here][9]][9]

  In mac the [app list][10] output not showed.

All the code works perfectly if I build it as an app.

Why this problem occured ?

  [1]: https://github.com/krogank9/WifiMouseServer
  [2]: https://i.stack.imgur.com/kwoVq.png
  [3]: https://github.com/krogank9/WifiMouseServer/blob/master/fakeinput-windows.cpp#L410
  [4]: https://github.com/krogank9/WifiMouseServer/blob/master/fakeinput-windows.cpp#L370
  [5]: https://github.com/krogank9/WifiMouseServer/blob/master/fakeinput-windows.cpp#L402
  [6]: https://github.com/krogank9/WifiMouseServer/blob/master/fakeinput-windows.cpp#L434
  [7]: https://i.stack.imgur.com/Adg9i.png
  [8]: https://github.com/krogank9/WifiMouseServer/blob/master/fakeinput-linux.cpp#L283
  [9]: https://i.stack.imgur.com/KQGdZ.png
  [10]: https://github.com/krogank9/WifiMouseServer/blob/master/fakeinput-mac.cpp#L305

  [11]: https://wifimouse.github.io/
  [12]: https://github.com/krogank9/WifiMouseServer
  [13]: https://github.com/krogank9/WifiMouseServer/blob/master/fakeinput-windows.cpp#L267
  [14]: https://i.stack.imgur.com/WiYAo.png

  [111]: https://github.com/krogank9/WifiMouseServer
  [112]: https://github.com/krogank9/WifiMouseServer/blob/master/WifiMouseServer.pro#L36
  [113]: https://github.com/nokeedev/gradle-native/discussions/639#discussion-4089091

**Comment by lacasseio** [comment]
Thanks for the question. Unfortunately, I fail to see how it relates to Gradle/Nokee. As much as I want to help you find a good solution to your problem, I don't want to set a bad precedent as a general build system related FAQs. Our focus is first and foremost the Nokee plugins for Gradle. I also support questions around Gradle core native support. The build for Wifimouse project uses qmake and make which is not something Nokee supports.

I can only echo what the [README](https://github.com/krogank9/WifiMouseServer/blob/master/README), more specifically the section about the dependencies. It would be best for this type of question to contact the original authors.

**Comment by pradeepsimba** [comment]
How to build qt shared library ?

[My project][1]

the app works perfectly if I build it as an app.

[TEMPLATE = app][2]

If I create a shared lib with same code the code not gives result.

I changed `TEMPLATE = app` to `TEMPLATE = lib` and I compile the same code. But , the result comes an error.

[code not give result as a qt shared lib][3]


  [1]: https://github.com/krogank9/WifiMouseServer
  [2]: https://github.com/krogank9/WifiMouseServer/blob/master/WifiMouseServer.pro#L36
  [3]: https://github.com/nokeedev/gradle-native/discussions/639#discussion-4089091

**Reply by lacasseio** [reply]
For this specific error, you would have to contact the [WifiMouseServer](https://github.com/krogank9/WifiMouseServer) project maintainer or open an issue on their project. This question is obviously not related to Nokee or Gradle so there is nothing I can do.

**Reply by lacasseio** [reply]
Just some additional information regarding your questions, I can see from your profile that you post the same question to a lot of unrelated projects. Although those projects (mine included) may have the knowledge to help you, the maintainers and their community are not a replacement for well searched and well throughout questions. The content of your questions is great, but they are mostly posted in the wrong place. Do the upfront research to determine a single place where the question makes the most sense in terms of context before posting. As mentioned by other members of the open-source community, the [project WifiMouseServer](https://github.com/krogank9/WifiMouseServer) is the best place for **this** question. Other questions will be best suited for a **different place**, like [StackOverflow](https://stackoverflow.com/) for example. Be mindful of that for your future questions.

**Reply by pradeepsimba** [reply]
I am really very very sorry !!!

**Reply by lacasseio** [reply]
No worries, don't feel bad. It's just something to be mindful of. Some people may be less tolerant. :-( If you ever have any questions related to Nokee or Gradle native support feel free to ask them here.
